oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -492,10 +503,10 @@ private static boolean isAwsCredential(Map<String, Object> credentialSource) {
   protected AccessToken exchangeExternalCredentialForAccessToken(
       StsTokenExchangeRequest stsTokenExchangeRequest) throws IOException {
     // Handle service account impersonation if necessary.
-    // Internal override takes priority.
-    if (impersonatedCredentialsOverride != null) {
-      return impersonatedCredentialsOverride.refreshAccessToken();
-    } else if (impersonatedCredentials != null) {
+    if (this.isImpersonationEnabled() && this.impersonatedCredentials == null) {
```

Comment:
Why we can we have a impersonationUrl but impersonated credentials not ready? Can we build them whenever url is set?

Reply:
see my reply on your first comment, building the impersonated credential in the parent constructor has some issues.